src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -320,20 +310,38 @@ public DocumentCollection createCollectionIfNotExists(@NonNull DocumentDbEntityI
         });
     }
 
-    public void deleteById(String collectionName, Object id, com.microsoft.azure.documentdb.PartitionKey partitionKey) {
+    @Override
+    public void deleteById(@NonNull String collectionName, @NonNull Object id, @Nullable PartitionKey partitionKey) {
         Assert.hasText(collectionName, "collectionName should not be null, empty or only whitespaces");
         assertValidId(id);
 
-        log.debug("execute deleteById in database {} collection {}", this.dbName, collectionName);
-
         try {
-            getDocumentClient().deleteDocument(getDocumentLink(this.dbName, collectionName, id.toString()),
-                    getDocumentDbRequestOptions(partitionKey, null));
-        } catch (com.microsoft.azure.documentdb.DocumentClientException ex) {
-            throw new DocumentDBAccessException("deleteById exception", ex);
+            deleteByIdAsync(collectionName, id, partitionKey).toBlocking().single();
+        } catch (RuntimeException e) {
+            final Throwable cause = e.getCause();
+
+            if (!(cause instanceof DocumentClientException)
+                    || ((DocumentClientException) cause).getStatusCode() != HttpStatus.SC_NOT_FOUND) {
+                throw new DocumentDBAccessException("Failed to delete Document", e);
```

Comment:
If the exception cause is not `DocumentClientException` or not `SC_NOT_FOUND` status code, the exception will be swallowed directly? Is it expected behavior?

Reply:
yes, if DocumentClientException and status code is 404. It should be no exist, for aligning with findById, eat the exception here.

Reply:
refactored the condition.